pages/measurement-sets/[id].js

```diff
@@ -169,6 +153,29 @@ export default function MeasurementSet({
       : [];
   const uniqueSampleSummaries = [...new Set(sampleSummaries)];
 
+  // Collect all measurement set protocols
+  let measurementSetProtocols = [];
+  if (measurementSet.protocols && measurementSet.protocols.length > 0) {
+    measurementSetProtocols = measurementSet.protocols;
+  }
+
+  // Collect all sample protocols.
+  const sampleProtocols = [];
+  if (samples.length > 0) {
+    samples.forEach((sample) => {
+      if (sample.protocols && sample.protocols.length > 0) {
+        sampleProtocols.push(...sample.protocols);
+      }
+    });
+  }
+  const uniqueSampleProtocols = [...new Set(sampleProtocols)];
+
+  // Combine measurement set and sample protocols.
+  const combinedProtocols = [].concat(
+    measurementSetProtocols,
+    uniqueSampleProtocols
+  );
```

Comment:
You can do:
```
const combinedProtocols = measurementSetProtocols.concat(uniqueSampleProtocols);
```
Or you could even not use `measurementSetProtocols` with:
```
const combinedProtocols = uniqueSampleProtocols.concat(measurementSet.protocols || []);
```
Is there any chance `measurementSet.protocols` and `uniqueSampleProtocols` could include the same URLs? I’m wondering if you need to deduplicate the combined list instead of deduplicating the samples list.

Reply:
i deduplicated the combined list and removed the code for deduplicating the samples list.